pkg/servicemesh/federation/discovery/controller.go

```diff
@@ -370,6 +375,30 @@ func (opt Options) validate() error {
 	return utilerrors.NewAggregate(allErrors)
 }
 
+func (c *Controller) getRootCertForMesh(instance *v1.MeshFederation) (string, error) {
+	if instance == nil {
+		return "", nil
+	}
+	name := instance.Spec.Security.CertificateChain.Name
+	if name == "" {
+		name = common.DefaultFederationCARootResourceName(instance)
+	}
+	entryKey := common.DefaultFederationRootCertName
+	switch instance.Spec.Security.CertificateChain.Kind {
```

Comment:
Is it guaranteed every component in `.Spec.Security.CertificateChain.Kind` is not null?

Reply:
It's a required field.  Must be specified by the user.